bluemira/base/tools.py

```diff
@@ -22,11 +22,36 @@
 """
 Tool function and classes for the bluemira base module.
 """
+import time
+from typing import Callable
+
 from bluemira.base.components import Component, PhysicalComponent
+from bluemira.base.look_and_feel import bluemira_debug, bluemira_print
 from bluemira.geometry.compound import BluemiraCompound
 from bluemira.geometry.tools import serialize_shape
 
 
+def timing(
+    func: Callable, timing_str: str, print_str: str = "", print_name: bool = True
+) -> Callable:
+    """Time a function and push to logging"""
```

Comment:
This is part of our public API (should it be?), so I think the arguments could do with being documented. It's not immediately obvious to me what they're doing from their names.

Reply:
I've documented and remove it from the public api also renamed the variables that hopefully make more sense